### ISSUE TITLE
Enhance user experience regarding secure connection requirements

### ISSUE THREAD

**Issue by dogeared** [stormpath/stormpath-sdk-java#409]
For any connection other than localhost, Stormpath requires a secure connection.

This may not be clear to people deploying to non-SSL protected environments.

This could result in confusion when code that was working locally suddenly stops working in a deployment environment.

**Comment by lhazlewood** [comment]
@dogeared do you have more information on this or what should change?

It is already covered (pretty deeply) in the login documentation in both the Servlet and Spring Boot docs:

https://docs.stormpath.com/java/servlet-plugin/login.html#security-notice
https://docs.stormpath.com/java/spring-boot-web/login.html#security-notice

Again, I'm not sure this should be in the quickstart because the quickstart (and probably even tutorial) should assume http://localhost:8080

If you still believe we should add more information around this (which I'd be fine with), do you have ideas on where it should go in the docs?

**Comment by dogeared** [comment]
Fair points. This issue originated from getting this in support a few times  over the last 6 months.

Perhaps we should have something in the README.md in the examples and tutorial code?

At least two of the support issues (maybe more) that I fielded were people taking one of the examples or tutorials, making some changes, testing locally and then throwing it into their AWS environment or other deployment environment and then having it not work.

There may be a deeper issue around this in that what happens in these cases is that the end result is that Account cookie doesn't get set. There's no error of any kind and so it's not entirely clear to the developer what's not working.

**Comment by mrioan** [comment]
The way I understand it, the "Security Notice" seems more a suggestion than a requirement. Anyways, supposing that people do understand it as a requirement they are for some reason not reading/finding it. I have had at least 5 support tickets dealing with HTTPS issues when moving to staging/prod/. I think we should do 2 things:

1) Improve the security notice. I would write something like what I pasted below.
2) Add a link in the quick start to the security notice so people are also aware that there is additional detailed documentation available

Security Notice:

"Because of the security requirements around identity cookies, the default implementation always requires HTTPS *except* when it has been determined that the request is sent and received by `localhost`.  This allows you to develop on your local machine without requiring a TLS/SSL certificate, but will require one when you deploy to production due to the security attack vectors that would occur otherwise.
Therefore, if you want to use anything other than localhost you will need to enable HTTPS for your Application."

**Comment by lhazlewood** [comment]
Maybe this is easily solved by inspecting the inbound request and if the request host is not localhost, we print a WARN message once and only once (so it doesn't keep happening) to let them know what is going on?

This is better IMO than documentation changes because they are likely to see it when running the application.  Thoughts?

**Comment by dogeared** [comment]
You mean a message to the log? I think that would be good and helpful at a minimum. I wonder if it should be a failed login attempt as well, though?

Documentation wise, perhaps we should change it from a "caution" to an "important" or a "danger" sphinx admonition type?

**Comment by lhazlewood** [comment]
From this mornings meeting:  We'll leave the 'caution' (yellow) admonitions as they are (they explicitly say that HTTPS is required in the first line).  We'll add an informative log `WARN` message however.
